src/Components/Forms/src/EditContextDataAnnotationsExtensions.cs

```diff
@@ -136,8 +155,125 @@ private void OnValidationRequested(object? sender, ValidationRequestedEventArgs
                     _messages.Add(new FieldIdentifier(_editContext.Model, fieldName: string.Empty), validationResult.ErrorMessage!);
                 }
             }
+        }
 
-            _editContext.NotifyValidationStateChanged();
+#pragma warning disable ASP0029 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+        private bool TryValidateTypeInfo(ValidationContext validationContext)
+        {
+            var options = _serviceProvider?.GetService<IOptions<ValidationOptions>>()?.Value;
```

Comment:
I might be wrong, but I don't think we need the null check here.

I believe IOptions<T> will always resolve a non-null value (we can assume that services.AddOptions has been called as its a dependency we have). I also believe that `.Value` will always be populated with a default instance